test/cmocka/src/math/dct/mfcc_get_dct_matrix.m

```diff
@@ -0,0 +1,53 @@
+% dct_matrix = mfcc_get_dct_matrix(num_ceps, num_mels, type)
+%
+% Input
+%   num_ceps - number of cepstral coefficients
+%   num_mels - number of Mel band energies
+%   type - DCT type, currently only 2 for DCT-II is supported
+%
+% Output
+%   dct matrix coefficients
+
+% SPDX-License-Identifier: BSD-3-Clause
+%
+% Copyright (c) 2022, Intel Corporation. All rights reserved.
+
+% Ref: pytorch kaldi.py _get_dct_matrix
+function dct_matrix = mfcc_get_dct_matrix(num_ceps, num_mels, type)
+
+dct_matrix = create_dct(num_ceps, num_mels, type, 'ortho');
+
+% make first cepstral to be weighted sum of factor sqrt(1 / num_mels)
+%dct_matrix(:,1) = sqrt(1 / num_mels);
+%dct_matrix = dct_matrix(:, 1:num_ceps);
+
+end
+
+
+% Ref pytorch functional.py create_dct
+function dct = create_dct(n_mfcc, n_mels, type, norm)
+
+n = (0:(n_mels - 1))'; % Size n_mels x 1
+k = 0:(n_mfcc - 1); % Size 1 x n_mfcc
+switch type
+	case 2,
+		arg = pi / n_mels * (n + 0.5) * k;
+		%max_arg = max(max(arg))
+		dct = cos(arg);
+	case 3,
+		error('not implemented'); % FIXME
+	otherwise
+		error('Unknown DCT type');
+end
+
+switch lower(norm)
+	case 'ortho',
+		dct(:,1) = dct(:,1) * 1 / sqrt(2);
+		dct = dct * sqrt(2 / n_mels);
+	case 'none'
+		dct = dct * 2;
```

Comment:
what this related case means ?

Reply:
Good question, I don't remember right now. It's there in Pytorch, check above link. I'll check if the Wikipedia equation for DCT-II if it's there of if it's something else.

Reply:
I found from this equation https://ccrma.stanford.edu/~jos/st/Discrete_Cosine_Transform_DCT.html the multiply by 2.  _"Thus, the real part of a double-length FFT is the same as the DCT except for the half-sample phase shift in the sinusoidal basis functions $ s_k(n) = e^{j\omega_k n}$ (and a scaling by 2 which is unimportant)."_ Maybe it's some scaling convention that happens to be endorsed into Kaldi and Pytorch?

Reply:
Thank you for your extra effort!
Then, in my opinion, you may just leave the `case 'none'` as-is.

Reply:
yep